MapboxDirections/Match/MBMatchingOptions.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+
+@objc(MBMatchingOptions)
+open class MatchingOptions: DirectionsOptions {
```

Comment:
Per chat, did you want to rename this class back to MatchOptions, for consistency with RouteOptions?

/ref https://github.com/mapbox/MapboxDirections.swift/pull/236#discussion_r163955397